pynn_spinnaker/spinnaker/regions/synaptic_matrix.py

```diff
@@ -240,21 +240,21 @@ def _write_spinnaker_row(self, row, next_row, next_row_offset,
             destination[2] = get_row_offset_length(next_row_offset,
                                                    len(next_row[1]),
                                                    self.LengthBits)
-
-        # Convert row to numpy record array
-        row = np.asarray(row[1], dtype=row_dtype)
-
-        # Extract the DTCM component of delay
-        # **NOTE** subtract one so there is a minimum of 1 slot of delay
-        dtcm_delay = 1 + ((row["delay"] - 1) % self.max_dtcm_delay_slots)
-
-        # Convert weight to fixed point
-        weight_fixed = float_to_weight(row["weight"])
-
-        # How much should we shift weights to be above index and delay
-        weight_shift = self.IndexBits + self.DelayBits
-
-        # Write row
-        destination[3:3 + len(row)] = (row["index"]
-                                       | (dtcm_delay << self.IndexBits)
-                                       | (weight_fixed << weight_shift))
+        # If there are any synapses in row
+        # **NOTE** empty rows may be tuples or empty
+        # lists both of which break the following code
+        if destination[0] > 0:
+            # Extract the DTCM component of delay
+            # **NOTE** subtract one so there is a minimum of 1 slot of delay
+            dtcm_delay = 1 + ((row[1]["delay"] - 1) % self.max_dtcm_delay_slots)
+
+            # Convert weight to fixed point
+            weight_fixed = float_to_weight(row[1]["weight"])
```

Comment:
Yes it does (it's your NumpyFloatToFixConverter from Rig)

I agree that Python is far from ideal for this...